Initial Snyk Module Submission

shawnsnyk opened the pull request:
Snyk Module

christopherhein commented on modules/snyk/snyk.adoc:
```diff
@@ -0,0 +1,295 @@
+
+Now in CodeBuild, look at your build history. Note it may take a minute or two for the new scan to run.
+
+image::snyk_4_Build.png[snyk]
```
Is this screenshot supposed to be fairly small and unreadable?

shawnsnyk replied:
Hi Chris, nice catch, somewhere along the process it was shrunk. I'll fix asap

christopherhein replied:
:+1:

christopherhein commented on modules/snyk/snyk.adoc:
```diff
@@ -0,0 +1,295 @@
+
+Let us pretend the developer fixed it and checked it in, coming back into the pipeline. This is done by changing 
+
+~/environment/aws-modernization-workshop/modules/containerize-application/app/pom.xml
```
Can we put this in back ticks?

christopherhein commented on modules/snyk/snyk.adoc:
```diff
@@ -0,0 +1,295 @@
+Push
+[source,shell]
+----
+git push -f codecommit master
```
I'm typically not a fan of and advise against anyone using `-f` any chance te can changer this to `--force-with-lease` still will force push but will verify that we're not overwriting someones additions to the mainline.

shawnsnyk replied:
actually just copied this from the automation exercise.. 
https://modernize.awsworkshop.io/index.html#/workshop/modernization-august-2018/module/modules_cicd_readme

I'm happy to use whatever you suggest. I kept that for consistency between labs. Can you confirm?

christopherhein replied:
Ah, we should change it in both places IMO … @jayyeras unless there is a good reason to keep `-f` vs `--force-with-lease`

jayyeras replied:
That appears to be an oversight on my part. I will go ahead and update the modules to show the correct usage.

christopherhein commented on modules/snyk/snyk.adoc:
```diff
@@ -0,0 +1,295 @@
+
+This time check code builder and we see it succeeded.
+
+image::snyk_4b_Build.png[snyk]
```
This again is fairly small, expected?

shawnsnyk replied:
Agreed. Same image issue as above

christopherhein commented on modules/snyk/snyk.adoc:
```diff
@@ -0,0 +1,295 @@
+   * Click View Report
+   * Set frequency project will be checked for vulnerabilities with the drop down list
+   * Click on View Report->Dependencies to see what libraries were used
+   * Use Integrations to connect and automate creation of fixes against a code repository
```
Through all these bullets I would make the important parts **bold** just so it's not lost, for example **Snyk.io** "Click **View Report**" and " **View Report** -> **Dependencies**"

shawnsnyk replied:
Agreed. I also want to use some color to make some lines stand out more. Let me look at that again. Have more time today to get fancy :)

christopherhein replied:
Cool, without customizing workshopper (which you totally can https://github.com/christopherhein/workshopper) I'm not sure how this you would apply it, so I'm looking forward to that if it's possible :)

shawnsnyk replied:
Hi @christopherhein , did you want to join the slack channel?
Make realtime easier? Just send me your email address.
In short, https://snyk.io/docs/using-snyk
You can use password same way we did in lab passing it to Dockerfile as an argument like we did in buildspec.yaml. 
After you do "bundler install" in Dockerfile you'd use "snyk test", it will look at your gemfile.lock.
Can setup a zoom if you're free now.. or we can use aws. Won't post contact details on public github.. Will ask Jay for your details.

shawnsnyk replied:
@christopherhein just wanted to check if you saw this and if you wanted to chat about above

christopherhein replied:
@shawnsnyk I think we're okay with this module. Let's sync after Re:invent to discuss workshoper, sound good?

shawnsnyk replied:
Sounds good sir... @christopherhein feel free to ping me anytime. @jayyeras can provide my contact info to contact me direct or add you to our shared slack. Have fun at Re:invent!

shawnsnyk commented:
I've updated the images, change the links to be relative, added formatting.
The one change I didn't make is with regards to the commit, as it's consistent with the aforementioned exercise.

christopherhein commented:
Awesome, thanks @shawnsnyk - I've pinged @jayyeras to confirm that we can change it in both places

christopherhein commented:
@jayyeras have anything else? I'm good with where we are.
